src/validation_mode.rs

```diff
@@ -0,0 +1,45 @@
+#[non_exhaustive]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+/// The preferred mode of validation for struct (de)serialization.
+/// It also affects which format is used by the client when sending queries.
+///
+/// - [`ValidationMode::First`] enables validation _only for the first `N` rows_
+///   emitted by a cursor. For the following rows, validation is skipped.
+///   Format: `RowBinaryWithNamesAndTypes`.
+/// - [`ValidationMode::Each`] enables validation _for all rows_ emitted by a cursor.
+///   This is the slowest mode. Format: `RowBinaryWithNamesAndTypes`.
+///
```

Comment:
Let's explicitly note that to avoid validation totally to have maximum performance, one should provide `ValidationMode::First(0)`